tests/components/webostv/test_notify.py

```diff
@@ -0,0 +1,121 @@
+
+from homeassistant.components.notify import ATTR_MESSAGE, DOMAIN as NOTIFY_DOMAIN
+from homeassistant.components.webostv import DOMAIN
+from homeassistant.const import CONF_ICON, CONF_SERVICE_DATA
+from homeassistant.helpers import discovery
+
+from . import TV_NAME, setup_webostv
+
+ICON_PATH = "/some/path"
+MESSAGE = "one, two, testing, testing"
+
+
+async def test_notify(hass, client):
+    """Test sending a message."""
+    await setup_webostv(hass, "fake-uuid")
+    assert hass.services.has_service(NOTIFY_DOMAIN, TV_NAME)
+
+    await hass.services.async_call(
+        NOTIFY_DOMAIN,
+        TV_NAME,
+        {
+            ATTR_MESSAGE: MESSAGE,
+            CONF_SERVICE_DATA: {
+                CONF_ICON: ICON_PATH,
+            },
+        },
+        blocking=True,
+    )
+    assert client.mock_calls[0] == call.connect()
+    assert client.connect.call_count == 1
+    client.send_message.assert_called_with(MESSAGE, icon_path=ICON_PATH)
+
+
+async def test_notify_not_connected(hass, client, monkeypatch):
+    """Test sending a message when client is not connected."""
+    await setup_webostv(hass, "fake-uuid")
+    assert hass.services.has_service(NOTIFY_DOMAIN, TV_NAME)
+
+    monkeypatch.setattr(client, "is_connected", Mock(return_value=False))
+    await hass.services.async_call(
+        NOTIFY_DOMAIN,
+        TV_NAME,
+        {
+            ATTR_MESSAGE: MESSAGE,
+            CONF_SERVICE_DATA: {
+                CONF_ICON: ICON_PATH,
+            },
+        },
+        blocking=True,
+    )
+    assert client.mock_calls[0] == call.connect()
+    assert client.connect.call_count == 2
+    client.send_message.assert_called_with(MESSAGE, icon_path=ICON_PATH)
+
+
+async def test_icon_not_found(hass, caplog, client, monkeypatch):
+    """Test notify icon not found error."""
+    await setup_webostv(hass, "fake-uuid")
+    assert hass.services.has_service(NOTIFY_DOMAIN, TV_NAME)
+
+    monkeypatch.setattr(client, "send_message", Mock(side_effect=FileNotFoundError))
+    await hass.services.async_call(
+        NOTIFY_DOMAIN,
+        TV_NAME,
+        {
+            ATTR_MESSAGE: MESSAGE,
+            CONF_SERVICE_DATA: {
+                CONF_ICON: ICON_PATH,
+            },
+        },
+        blocking=True,
+    )
+    assert client.mock_calls[0] == call.connect()
+    assert client.connect.call_count == 1
+    client.send_message.assert_called_with(MESSAGE, icon_path=ICON_PATH)
+    assert f"Icon {ICON_PATH} not found" in caplog.text
+
+
+@pytest.mark.parametrize(
+    "side_effect,error",
+    [
+        (WebOsTvPairError, "Pairing with TV failed"),
+        (ConnectionRefusedError, "TV unreachable"),
+    ],
+)
+async def test_connection_errors(hass, caplog, client, monkeypatch, side_effect, error):
+    """Test connection errors scenarios."""
+    await setup_webostv(hass, "fake-uuid")
+    assert hass.services.has_service("notify", TV_NAME)
+
+    monkeypatch.setattr(client, "is_connected", Mock(side_effect=side_effect))
```

Comment:
`is_connected` doesn't have any side effects in the code under test. It's a bit weird that we apply side effects on that. It's more correct to set the side effects on client.connect.